lib/src/widgets/chat/ds_url_preview.widget.dart

```diff
@@ -0,0 +1,137 @@
+import 'package:blip_ds/blip_ds.dart';
+import 'package:blip_ds/src/controllers/chat/ds_url_preview.controller.dart';
+import 'package:flutter/material.dart';
+import 'package:get/get.dart';
+import 'package:url_launcher/url_launcher.dart';
+
+/// A Design System's clickable URL previewer that shows some metadata infos about the given [url].
+class DSUrlPreview extends StatelessWidget {
+  final Uri url;
+  final Color backgroundColor;
+  final Color foregroundColor;
+  final List<DSBorderRadius> borderRadius;
+  final EdgeInsets bodyPadding;
+
+  final _controller = DSUrlPreviewController();
+
+  /// Creates a Design System's clickable URL previewer that shows some metadata infos about the given [url].
+  DSUrlPreview({
+    required this.url,
+    required this.backgroundColor,
+    required this.foregroundColor,
+    super.key,
+    this.borderRadius = const [DSBorderRadius.all],
+    this.bodyPadding = const EdgeInsets.symmetric(
+      vertical: 8.0,
+      horizontal: 16.0,
+    ),
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    _controller.getUrlPreview(url);
+
+    return Obx(
+      () => _controller.urlPreview.value?.url == null &&
+              !_controller.isLoading.value
+          ? const SizedBox()
+          : Padding(
+              padding: const EdgeInsets.all(4.0),
+              child: DSAnimatedSize(
+                child: Container(
+                  width: double.infinity,
+                  clipBehavior: Clip.hardEdge,
+                  decoration: BoxDecoration(
+                    borderRadius: borderRadius.getCircularBorderRadius(
+                      maxRadius: 18.0,
+                    ),
+                    color: backgroundColor,
+                  ),
+                  child: _controller.isLoading.value
+                      ? _buildLoading()
+                      : _buildPreview(
```

Comment:
I think you could use `_controller.urlPreview `inside the `_buildPreview` method, so you don't need to pass parameters to `_buildPreview`